libs/community/langchain_community/vectorstores/sqlserver.py

```diff
@@ -89,6 +120,24 @@ class EmbeddingStore(Base):
     def embeddings(self) -> Embeddings:
         return self.embedding_function
 
+    @property
+    def distance_strategy(self) -> str:
+        # Value of distance strategy passed in should be one of the supported values.
+        if isinstance(self._distance_strategy, DistanceStrategy):
+            return self._distance_strategy.value
+
+        # Match string value with appropriate enum value, if supported.
+        distance_strategy_lower = str.lower(self._distance_strategy)
+
+        if distance_strategy_lower == DistanceStrategy.EUCLIDEAN.value:
+            return DistanceStrategy.EUCLIDEAN.value
```

Comment:
will the return always be converted to lower case? Is that mandatory or just a choice?

Reply:
Well, take Euclidean as a sample case, SQLServer requires that it be passed in as `EUCLIDEAN` or `euclidean`. It won't recognize a mixed case distance strategy.

Reply:
if the SQLServer has a CASE-sensitive collation, will always providing lowercase work? Do we instead need this to be always be an uppercase? Can you please test it out?

Reply:
Confirmed this on a DB with case sensitive collation - `SQL_Latin1_General_CP1_CS_AS`. It works as expected.

Reply:
Is there any way to add a regression test for various collations?